tuxemon/core/event/conditions/party_level.py

```diff
@@ -0,0 +1,94 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+
+from tuxemon.core.event.eventcondition import EventCondition
+
+logger = logging.getLogger(__name__)
+
+
+class PartyLevelCondition(EventCondition):
+    """ Checks to see where an NPC is facing
+    """
+    name = "party_level"
+
+    def test(self, game, condition):
+        """Perform various checks about the player's party size. With this condition you can see if
+        the player's party is less than, greater than, or equal to then number you specify.
+
+        :param game: The main game object that contains all the game's variables.
+        :param condition: A dictionary of condition details. See :py:func:`core.map.Map.loadevents`
+            for the format of the dictionary.
+
+        :type game: core.control.Control
+        :type condition: Dictionary
+
+        :rtype: Boolean
+        :returns: True or False
+
+        Valid Parameters: check, level
+
+        The "check" parameter can be one of the following: "average", "lowest", or "highest".
+        """
+        check = str(condition.parameters[0])
+        level = int(condition.parameters[1])
+
+        """
+        player = game.player1
+        for monster in player.monsters:
+            
+        """
+        if len(game.player1.monsters) == 0:
+            return False
+
+        level_lowest = 100
+        level_highest = 0
+        level_average = 0
+        for monster in game.player1.monsters:
+            if monster.level < level_lowest:
+                level_lowest = monster.level
+            if monster.level > level_highest:
+                level_highest = monster.level
+            level_average += monster.level
+        level_average = int(round(level_average / len(game.player1.monsters)))
+
+        # Check to see if the party's average level is beyond this level.
+        if check == "average":
+            return level_average >= level
+
+        # Check to see if the party's lowest level monster has reached this level.
+        elif check == "lowest":
+            return level_lowest >= level
+
+        # Check to see if the party's highest level monster has reached this level.
+        elif check == "highest":
+            return level_highest >= level
+
```

Comment:
suggestion: store the lowest, highest, and average levels as variables so that a dialog can use them.  For instance "You cannot join this tournament because your highest level monster is only X!"

Reply:
For that to work properly though, you would have to run this action, then a dialog after it.

Reply:
That's actually a very good idea! Only problem with it is that, it would require a new action, and I'm not sure if adding one just for this makes sense. It would be somewhat absurd to set variables upon checking a conditional. So not sure how to go about this one.

Reply:
i see what you are saying.  actually, since i am thinking about it, a condition runs at least once per frame, so the variables will always be updated.  as long as the map has this condition, then these variables will be up to date.  i was mistaken by saying "run this action first"....since its a condition, the game is checking it often.

Reply:
its maybe a potential bug if a map designer doesn't add this condition if they want these values in a dialog, but i think its a very easy fix and they likely will have this condition on the map if they have a dialog about it, so i don't see a huge problem.

Reply:
Actually, in this case, I can simply remove the conditional and do this as an action entirely. You can simply check the variables directly then, no need for extra functions.

In fact... I could do this internally altogether! Whenever a monster is added or levels up, the game variables with the party levels are automatically updated, no need for any more map actions. Should I go with this approach?

Reply:
thats not a bad idea.  its somewhat problematic because monsters would have to have a reference to their player object.  monsters shouldn't have party logic in them.  npcs shouldn't be directly modifying game variables because variables are for the game player.  suppose a trainer npc levels up a monster, then it would overwrite the game variables.  i do think your idea is best overall, but i think it would require more changes to do it correctly, and the game isn't really at a point to support it.  maybe adding a conditional 'if self is game.player1` might work.  feel free to experiment.

Reply:
The latest commit contains what seemed like the best compromise solution: An NPC function records party status variables now. This function is filtered by self.is_player so running it for another NPC will make no changes.